### ISSUE TITLE
[MRESOURCES-132] Copying of files with permissions broken

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-resources-plugin#260]
**[Martin Todorov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=carlspring)** opened **[MRESOURCES-132](https://issues.apache.org/jira/browse/MRESOURCES-132?redirect=false)** and commented

Files with permissions are not copied with the permissions as illustrated below:

```
root@carlspring:/java/opensource/build-force/zipper-plugin# ls -al `find . -name *perm*`
-rwxr--r-- 1 root root   8 2010-12-09 20:48 ./src/test/resources/zip/file.with.permission.sh*
-rw-r--r-- 1 root root   8 2010-12-09 22:56 ./target/test-classes/zip/file.with.permission.sh
```

Notice the 'x' permission missing from the copied file.

---

**Affects:** 2.4.3

7 votes, 26 watchers

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

Long time no supplemental information...Don't hesitate to reopen if you have other related information or may best a sample project which shows the wrong behaviour...

**Comment by jira-importer** [comment]
**[Martin d'Anjou](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=martinda)** commented

I have isolated the problem and can reproduce it easily. I use Maven 3.0.5, and Java 8. The steps to reproduce are:

```
git clone https://github.com/martinda/maven-resource-permission-bug-132
cd maven-resource-permission-bug-132
mvn test
```

The test fails to execute the `runme` script. The permissions were not copied by maven, resulting in this error:

```
java.io.IOException: Cannot run program "/home/martin/tmp/maven-resource-permission-bug-132/target/test-classes/runme": error=13, Permission denied
```

If I force the correct permissions by hand, the test passes:

```
chmod +x target/test-classes/runme
mvn test
```

I do not seem to have permissions to reopen the bug, could you do that for me please? Thanks.

**Comment by jira-importer** [comment]
**[Nikolai Grigoriev](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ngrigoriev@gmail.com)** commented

My build also suffers from this problem. As simple as that:

```
<plugin>
  <artifactId>maven-resources-plugin</artifactId>
  <version>2.7</version>
  <executions>
    <execution>
      <id>copy-resources</id>
      <phase>validate</phase>
      <goals>
        <goal>copy-resources</goal>
      </goals>
      <configuration>
        <outputDirectory>${basedir}/target/scripts</outputDirectory>
        <resources>
          <resource>
              <directory>${basedir}/src/main/scripts</directory>
              <filtering>false</filtering>
          </resource>
        </resources>
      </configuration>
    </execution>
  </executions>
</plugin>
```

I have an executable script that loses its executable permissions after being copied with this plugin.

**Comment by jira-importer** [comment]
**[Gili](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cowwoc)** commented

[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise) Please reopen this issue. Recent comments provide new repro steps and I see no indication that this bug is fixed in the latest version.

**Comment by jira-importer** [comment]
**[Hassan Khan](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hassankhan)** commented

Please reopen this issue. I am getting the same issue and can reproduce on my local. This is high priority item.

**Comment by jira-importer** [comment]
**[Mohsen Sarmadi](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=msarmadi)** commented

I have this issue too, can you reopen this ticket

Thanks

**Comment by jira-importer** [comment]
**[Aleksander Rykalin](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=arykalin)** commented

It's still there. I understand the point about I\O but can you add an option to copy permissions as well?

**Comment by jira-importer** [comment]
**[OLIVIER GERARDIN](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ogerardin)** commented

Issue still present in latest version (3.1.0). Must use a hack with `maven-antrun-plugin` to perform a "chmod" before running a script that was copied as a resource. The script is filtered so running the original one is not an option.

Please reopen this issue and provide an option to preserve file permissions.

**Comment by jira-importer** [comment]
**[Bernd Schatz](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=beschat)** commented

Issue is still present with 3.3.9

The issue is old and very easy to reproduce, why was this BUG closed ?

please reopen it.

**Comment by jira-importer** [comment]
**[Daniele Pirola](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=daniele.pirola@icteam.it)** commented

Issue still present with plugin v 3.1.0

I don't undestand why this issue was closed and why was closed with "cannot reproduce".

Also MRESOURCES-236 was closed with "Fixed" but I can't find which version of the plugin fixed this issue.

Please reopen and fix

**Comment by jira-importer** [comment]
**[Martin Todorov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=carlspring)** commented

[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise), [Olivier Lamy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=olamy),

 

Could you please re-open this issue?

**Comment by jira-importer** [comment]
**[Dennis Lundberg](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dennisl)** commented

MRESOURCES-236 has been fixed and will be included in the upcoming version 3.2.0.
If you have the possiblility, please try 3.2.0-SNAPSHOT.
If you have comments, please do so in MRESOURCES-236.
